scripts/pylib/twister/twisterlib.py

```diff
@@ -3311,6 +3311,13 @@ def xunit_report(self, filename, platform=None, full_report=False, append=False,
                 handler_time = instance.metrics.get('handler_time', 0)
                 duration += handler_time
                 if full_report and instance.run:
+                    # sometimes a test instance hasn't been executed successfully with an
```

Comment:
this is the wrong place to do that, it should not be set for junit reports only and instead be available to any type of reporting we might generate, so this basically needs to be done at the execution and test result evaluation level.

Reply:
moved to handle function of DeviceHandler.